clients/views.py

```diff
@@ -2,149 +2,158 @@
+        clients = Client.objects.filter(company=self.request.user.company)
+        query = self.request.GET.get("q")
+        if query:
+            clients = clients.filter(Q(display_name__icontains=query,  company=self.request.user.company) )
+        context['clients'] =  clients
+        return context
+
+
+    def get(self, *args, **kwargs):
+        '''displaying the data of clients
+        '''
+        return render(self.request, self.template_name, self.get_context_data(**kwargs))
+
+
+
+class ClientView(LoginRequiredMixin,TemplateView):
+    '''display client information
+    '''
+    template_name = 'clients/view_client.html'
+
+
+    def get_context_data(self,**kwargs):
+        '''getting the client data from database
+        '''
+        context = super().get_context_data(**kwargs)
+        context['client'] = get_object_or_404(Client, id=kwargs['client_id'])
+        context['current_path'] = self.request.path
+        context['invoices'] = Invoice.objects.filter(client=kwargs['client_id'])
+        return context
+
+    
+    def get(self, *args, **kwargs):
+        '''displaying the data of client
+        '''
+        return render(self.request, self.template_name, context=self.get_context_data(**kwargs))
 
 
 
 class ClientAddView(LoginRequiredMixin,TemplateView):
-	template_name = 'clients/update_client.html'
+    '''Viewing for adding client
+    '''
+    template_name = 'clients/update_client.html'
 
-	def get(self, *args, **kwargs):
-		return render(self.request, self.template_name, context=self.get_context_data(**kwargs))
 
-	def post(self, *args, **kwargs):
-		#import pdb; pdb.set_trace()
-		client_form = ClientForm(self.request.POST,user=self.request.user)	
-		#import pdb; pdb.set_trace()
-		if client_form.is_valid() :
-			
-			client = client_form.save(commit=False)		
-			
-			client.save()
-			messages.success(self.request, 'Client is successfully added')
-			return redirect('clients')
-			
-		else:
-			context = {
-				'client_form' : client_form,
-				
+    def get(self, *args, **kwargs):
+        '''display the client form
+        '''
+        return render(self.request, self.template_name, context=self.get_context_data(**kwargs))
 
-				'client_form_errors' : client_form.errors,
 
-			}
-		return render(self.request, self.template_name, context=context)
+    def post(self, *args, **kwargs):
+        '''getting the filled form of client
+        '''
+        errors = {}
+        form = ClientForm(self.request.POST, user=self.request.user)  
+        if form.is_valid():
+            '''form validation for client
+            '''
+            client = form.save(commit=False, company=self.request.user.company)     
+            client.save()
+            messages.success(self.request, 'Client is successfully added')
+            return redirect('clients')
+        else:
+            context=self.get_context_data(**kwargs)
+            context['form'] = form
+            context['form_errors'] = form.errors
```

Comment:
this is not needed. based on the documentation you can get the errors from the form instance itself.

`{{ form.<fieldname>.errors.as_text }}`

`eg. <span class="error">{{ form.email.errors.as_text }}</span>`